### PR TITLE
Rebuild for hdf51122 fixed

### LINKED ISSUE #115: h5py 3.7.0 ARM not building
### Comment:

Currently the ARM builds still fail in CI, see https://github.com/conda-forge/h5py-feedstock/pull/114.

Maintainers merged the above PR to at least move forward with the green builds (linux x64, osx, windows).

Adding a list with the failing architectures after processing of https://github.com/conda-forge/h5py-feedstock/runs/7074185121.

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR augments #118 with the following fixes

- (bb08b911ba9e7eb8e4ff2c6731c71f14d3233c0f) don't cross-compile linux aarch64 or ppc64le
- (62612cb117c62b71f39ef9fb1d2a7a22babf63cf) set `CIBW_ARCHS_MACOS` for osx-arm64 (cherry-pick of https://github.com/conda-forge/h5py-feedstock/pull/117/commits/975240b33d4b3b9fb01e6d0a14f3b1170a9ba2b1 from #117)
- (fcbb1ff94f345513851e78190b839529ac8a087b) add `hdf5` to the build environment when cross compiling and tell the build script to use it

Closes #115
Closes #118

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@conda-forge/h5py, this is ready for review, I think (cc @hmaarrfk).

File: recipe/build.sh [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+else
+  export HDF5_DIR=${PREFIX}
+fi
 export OPAL_PREFIX=${PREFIX}
 if [[ "$mpi" != "nompi" ]]; then
   export HDF5_MPI="ON"
 fi
 
+if [[ ${target_platform} == "osx-arm64" ]]; then
+  export CIBW_ARCHS_MACOS=arm64
```
Comment (hmaarrfk) [inline recipe/build.sh:17]: I think this line deserves a comment explaining why we are using a flag supposedly targeting a different build system

Comment (hmaarrfk) [reply recipe/build.sh:17]: I even forgot why I added it....

**Comment by hmaarrfk** [comment]
I'm not sure why cross compiling isn't working for arch and ppc64le. It should in theory work.

Travis is quite slow (mostly due to our limited access to it) so for these large build matrices, maybe it is good to do them on azure.

**Review by hmaarrfk — APPROVED** [review]
Great job!

File: recipe/build.sh (hunk 1 above)
Comment (duncanmmacleod) [reply recipe/build.sh:17]: Happy to add anything you suggest (presuming you can't directly commit to this branch), how about:
```suggestion
  # disable complex256 on macOS ARM64, see https://github.com/h5py/h5py/pull/2065
  export CIBW_ARCHS_MACOS=arm64
```

Comment (hmaarrfk) [reply recipe/build.sh:17]: great!

**Comment by duncanmmacleod** [comment]
> I'm not sure why cross compiling isn't working for arch and ppc64le. It should in theory work.

I think this is because of the use of `ctypes.CDLL(...)` in `setup_configure.py`. This should be fixed by fcbb1ff94f345513851e78190b839529ac8a087b, so bb08b911ba9e7eb8e4ff2c6731c71f14d3233c0f could probably be harmlessly reverted if required.

**Comment by duncanmmacleod** [comment]
> Travis is quite slow (mostly due to our limited access to it) so for these large build matrices, maybe it is good to do them on azure.

From the tiny dataset that this PR represents, the travis-ci build matrix completes _faster_ than the azure build matrix. The linux-ppc64le builds are the slowest, but are still faster than the win-64 builds run on Azure. 

I appreciate that this is highly dependent on the total job queue on travis-ci, and UK morning/early-afternoon on a Sunday is probably the best time to attempt a large parallel matrix, but even if it is slower in general I prefer simplicity of configuration (using the 'default' provider in `conda-forge.yml`) over speed.

**Comment by hmaarrfk** [comment]
Thank you for the thorough discussion, I think that this is a fine PR and should be merged. I'll let it sit here until a maintainer sees it.

**Comment by hmaarrfk** [comment]
Lets move on. The discussion was quite clear that the intention was to use the previous as a crutch, while a real fix is found.

Thanks for following through.

**Pull request merged**